packages/amplify-category-function/src/provider-utils/awscloudformation/utils/layerHelpers.ts

```diff
@@ -36,8 +35,8 @@ export function layerNameQuestion(projectName: string): InputQuestion {
     validate: (input: string) => {
       input = input.trim();
       const meta = stateManager.getMeta();
-      if (!/^[a-zA-Z0-9-_]{1,87}$/.test(input)) {
-        return 'Lambda layer names must be 1-87 characters long. Only alphanumeric, -, and _ characters supported.';
+      if (!/^[a-zA-Z0-9]{1,87}$/.test(input)) {
```

Comment:
I see that now we restrict the names, but would not it cause a migration scenario for already deployed layers? Perhaps we could do a regex replace to remove the - now forbidden - characters?

Reply:
Luckily not, this is how the current implementation works. I mistakenly changed it thinking we could be less strict in a previous PR to the layers-rework branch, just reverting the change I made.